File: src/main/java/net/beams/genera/entities/render/RenderFaerie.java
package net.beams.genera.entities.render;

import net.beams.genera.entities.passive.EntityFaerie;
import net.beams.genera.entities.render.models.ModelFaerie;
import net.beams.genera.lib.Constants;
import net.minecraft.client.renderer.entity.Render;
import net.minecraft.client.renderer.entity.RenderLiving;
import net.minecraft.client.renderer.entity.RenderManager;
import net.minecraft.util.ResourceLocation;
import net.minecraftforge.fml.client.registry.IRenderFactory;

import javax.annotation.Nullable;

/**
 * Created by ben on 3/25/17.
 */
public class RenderFaerie extends RenderLiving<EntityFaerie> {

    private ResourceLocation texture = new ResourceLocation(Constants.MODID, "textures/entity/faerie.png");
    public static final Factory FACTORY = new Factory();

    public RenderFaerie(RenderManager rendermanagerIn) {
        super(rendermanagerIn, new ModelFaerie(), 0.2F);
    }

    @Nullable
    @Override
    protected ResourceLocation getEntityTexture(EntityFaerie entity) {
        return texture;
    }

    public static class Factory implements IRenderFactory<EntityFaerie> {
        @Override
        public Render<? super EntityFaerie> createRenderFor(RenderManager manager)  {
            return new RenderFaerie(manager);
        }
    }
}

File: src/main/kotlin/net/bms/genera/entity/render/RenderFaerie.kt
package net.bms.genera.entity.render

import net.bms.genera.Genera
import net.bms.genera.custom.DefaultFaeries
import net.bms.genera.entity.EntityFaerie
import net.minecraft.client.model.ModelBase
import net.minecraft.client.renderer.entity.RenderLiving
import net.minecraft.client.renderer.entity.RenderManager
import net.minecraft.util.ResourceLocation

class RenderFaerie(manager: RenderManager, model: ModelBase): RenderLiving<EntityFaerie>(manager, model, 0.2F) {
    override fun getEntityTexture(entity: EntityFaerie?): ResourceLocation? {
        return when (entity?.faerieInformation?.type) {
            DefaultFaeries.underground?.type -> ResourceLocation(Genera.MODID, "textures/entity/faerie/${DefaultFaeries.underground?.bodyTexture}.png")
            DefaultFaeries.woodland?.type -> ResourceLocation(Genera.MODID, "textures/entity/faerie/${DefaultFaeries.woodland?.bodyTexture}.png")
            DefaultFaeries.highland?.type -> ResourceLocation(Genera.MODID, "textures/entity/faerie/${DefaultFaeries.highland?.bodyTexture}.png")
            else -> ResourceLocation(Genera.MODID, "textures/entity/faerie/${DefaultFaeries.woodland?.bodyTexture}.png")
        }
    }
}